presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraMetadata.java

```diff
@@ -207,24 +212,31 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
         }
         else {
             CassandraClusteringPredicatesExtractor clusteringPredicatesExtractor = new CassandraClusteringPredicatesExtractor(
-                    cassandraSession.getTable(getTableName(handle)).getClusteringKeyColumns(),
+                    cassandraSession.getTable(handle.getSchemaTableName()).getClusteringKeyColumns(),
                     partitionResult.getUnenforcedConstraint(),
                     cassandraSession.getCassandraVersion());
             clusteringKeyPredicates = clusteringPredicatesExtractor.getClusteringKeyPredicates();
             unenforcedConstraint = clusteringPredicatesExtractor.getUnenforcedConstraints();
         }
 
-        ConnectorTableLayout layout = getTableLayout(session, new CassandraTableLayoutHandle(
-                handle,
+        CassandraTableHandle newHandle = new CassandraTableHandle(
```

Comment:
This could be structured a bit differently:

```java
if (handle.getPartitions().containsAll(partitionResult.getPartitions()) && handle.getClusteringKeyPredicates.equals(clusteringKeyPredicates) {
    return Optional.empty();
}

return Optional.of(new ConstraintApplicationResult<>(new CassandraTableHandle(...), unenforcedConstraint));
```